src/Microsoft.Data.Sqlite.Core/SqliteTransaction.cs

```diff
@@ -43,9 +43,10 @@ internal SqliteTransaction(SqliteConnection connection, IsolationLevel isolation
                 throw new ArgumentException(Resources.InvalidIsolationLevel(isolationLevel));
             }
 
-            // TODO: Register transaction hooks to detect when a user manually completes a transaction created using
-            //       this API
-            connection.ExecuteNonQuery("BEGIN;");
+            connection.ExecuteNonQuery(
+                IsolationLevel == IsolationLevel.Serializable
```

Comment:
That's why I'm not using the parameter `isolationLevel`, but instead the property `IsolationLevel`. This will check to see what the current isolation level is.